### PR TITLE
Explicit namespaces in sort-related docstrings

### PR DESCRIPTION
**Pull request by jishnub**
This represents the unexported names by the fully qualified namespace, which makes it simpler to interpret the keyword arguments. Otherwise, for example, it's not obvious where `Forward` comes from in the `sort` docstring.

Also, replaces some occurrences of `instance <: Algorithm` by `instance isa Algorithm`, which is the correct relation. This is also closer to English, which helps with readability.

Additionally, I think `Base.Sort.defalg` should be documented, as this is being used in the function signatures. This may be done separately.

### CODE REVIEW

File: base/sort.jl [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ end
 
 
 """
-    BracketedSort(target[, next::Algorithm]) <: Algorithm
+    BracketedSort(target[, next::Algorithm]) isa Algorithm
```
Comment (LilithHafner) [inline base/sort.jl:1160]:
```suggestion
    BracketedSort(target[, next::Algorithm]) isa Base.Sort.Algorithm
```
For consistency

File: base/sort.jl [hunk 2]
```diff
@@ -1346,7 +1346,7 @@ end
 
 
 """
-    StableCheckSorted(next) <: Algorithm
+    StableCheckSorted(next) isa Algorithm
```
Comment (LilithHafner) [inline base/sort.jl:1349]:
```suggestion
    StableCheckSorted(next) isa Base.Sort.Algorithm
```

**Review by LilithHafner — APPROVED** [review]
These seem like improvements

Changing `CountingSort <: Algorithm` to `CountingSort() isa Algorithm` would not be an improvement on its own, as the declaration of Counting sort is literally `struct CountingSort <: Algorithm end`, but I guess it makes sense for consistency.

This also documents the internals from a Base user's perspective which, absent the public keyword, would be a problem, but with that it's clear these are not public so it's okay to document them in a way that focuses on the experience of Base users poking around.

Overall, LGTM. Thanks!

**Comment by jishnub** [comment]
Personally I don't prefer the `CountingSort() isa Algorithm` over the current style either, but I did indeed change this for consistency. I suppose this form makes it clear that the constructor doesn't take any argument.

**Pull request merged**